src/app/forms/validators/date/date.validators.spec.ts

```diff
@@ -0,0 +1,20 @@
+import { FormControl } from '@angular/forms';
+import { DateValidators } from './date.validators';
+
+describe(DateValidators.validDate.name, () => {
+  it.each([
+    [{ invalidDate: true }, 'asdasd'],
+    [{ invalidDate: true }, '2000-01-00'],
+    [{ invalidDate: true }, '2000--00'],
+    [{ invalidDate: true }, '20'],
+    [{ invalidDate: true }, '-01-00'],
+    [{ invalidDate: true }, ''],
+    [null, '2022-01-01'],
+    [null, '2022-01'],
+    [null, '2022']
+  ])('should validate date and return %s for %p', (expected, date) => {
+    const control = new FormControl(new Date(date));
+
+    expect(DateValidators.validDate(control)).toEqual(expected);
+  });
```

Comment:
Is it possible to have tests that prove the value can be read and stored as an ISO-8601 timestamp? (given that's how we've seen the data)

Reply:
The value is propagated as Date object and should be converted to JSON string when we call the API. As for reading, the Date Pipe should handle the format as normal I think